functions/go/set-setter/setters/set.go

```diff
@@ -0,0 +1,153 @@
+package setters
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Set sets resource field values from an OpenAPI setter
+type Filter struct {
+	// Setters holds the user provided values for all the setters
+	Setters []Setter
+}
+
+type Setter struct {
+	// SetterName is the name of the setter
+	SetterName string `json:"setter_name,omitempty" yaml:"setter_name,omitempty"`
+
+	// SetterValue is the input value for setter
+	SetterValue interface{} `json:"setter_value,omitempty" yaml:"setter_value,omitempty"`
```

Comment:
Why `interface{}`? Any other possible types besides string?

Reply:
Yes. It could be list values. Which will be added in coming PRs.